testing/src/main/java/io/stargate/it/grpc/ExecuteBatchStreamingTest.java

```diff
@@ -0,0 +1,158 @@
+import io.stargate.grpc.Values;
+import io.stargate.it.driver.CqlSessionExtension;
+import io.stargate.it.driver.CqlSessionSpec;
+import io.stargate.it.driver.TestKeyspace;
+import io.stargate.proto.QueryOuterClass;
+import io.stargate.proto.QueryOuterClass.Batch;
+import io.stargate.proto.QueryOuterClass.Response;
+import io.stargate.proto.QueryOuterClass.ResultSet;
+import io.stargate.proto.StargateGrpc.StargateStub;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(CqlSessionExtension.class)
+@CqlSessionSpec(
+    initQueries = {
+      "CREATE TABLE IF NOT EXISTS test (k text, v int, PRIMARY KEY(k, v))",
+    })
+public class ExecuteBatchStreamingTest extends GrpcIntegrationTest {
+
+  @AfterEach
+  public void cleanup(CqlSession session) {
+    session.execute("TRUNCATE TABLE test");
+  }
+
+  @Test
+  public void simpleStreamingBatch(@TestKeyspace CqlIdentifier keyspace) {
+    List<Response> responses = new ArrayList<>();
+
+    StargateStub stub = asyncStubWithCallCredentials();
+    StreamObserver<Response> responseStreamObserver =
+        new StreamObserver<Response>() {
+          @Override
+          public void onNext(Response value) {
+            System.out.println("client onNext: " + value);
+            responses.add(value);
+          }
+
+          @Override
+          public void onError(Throwable t) {
+            System.out.println("error: " + t);
+          }
+
+          @Override
+          public void onCompleted() {
+            System.out.println("client completed ");
+          }
+        };
+
+    StreamObserver<Batch> requestObserver = stub.executeBatchStream(responseStreamObserver);
+
+    requestObserver.onNext(
+        Batch.newBuilder()
+            .addQueries(cqlBatchQuery("INSERT INTO test (k, v) VALUES ('a', 1)"))
+            .addQueries(
+                cqlBatchQuery(
+                    "INSERT INTO test (k, v) VALUES (?, ?)", Values.of("b"), Values.of(2)))
+            .setParameters(batchParameters(keyspace))
+            .build());
+
+    requestObserver.onNext(
+        Batch.newBuilder()
+            .addQueries(
+                cqlBatchQuery(
+                    "INSERT INTO test (k, v) VALUES (?, ?)", Values.of("c"), Values.of(3)))
+            .setParameters(batchParameters(keyspace))
+            .build());
+
+    // all inserted records may be not visible to the subsequent SELECT query
+    // because all reactive calls are non-blocking. Therefore, we need to wait for response of two
+    // batch insert
+    // queries
+    Awaitility.await()
+        .atMost(10, TimeUnit.SECONDS)
+        .until(
+            () -> {
+              return responses.size() == 2;
```

Comment:
if this is async, `responses` should be synced list, like `CopyOnWriteArratList`